### ISSUE TITLE
How to use

### ISSUE THREAD

**Issue by ghost** [pb-jchin/ASTER#1]
Hi Jason

How to use this tool?
What ist the input data (falcon?)? How to run (I mean how to fire it up and did I get it right hat I can combine it with IGV)?

so many questions... :-)

**Comment by pb-jchin** [comment]
yes.. yes.. so many question and this job never gets me bored... (I just sometimes do need to have a true vacation.) Be patient and thank in advance.

**Comment by pb-jchin** [comment]
Here is something that might help for now,

> It takes the FALCON assembly graph output (`sg_edge_list`, `utg_data`, and `ctg_paths` in the falcon output) as input.  The location of thesis input is hardcoded in the `graph_data_server.py`, so you might need to move the data into the directory of the script or modify the code to point the file into the right location. (I will modify this into configurable options someday.)
> 
> At the backend, one needs to start two web servers locally with two scripts (`graph_data_server.py` for storing large data in memory for long period of time and `main_loop.py` for process the requests other than the graph data). The front end (`https://github.com/pb-jchin/ASTER/blob/master/src/view/index.html` ) is executed by a browser. The code will try to connect to the server (`main_loop.py`) through “websocket” so it can respond to other requests to the same server. You will need to start to server first to ensure the web-socket is properly connected.
> 
> The latest developer build of IGV http://software.broadinstitute.org/software/igv/download_snapshot can make HTTP request. See discussion on github:
> 
> https://github.com/igvteam/igv/issues/277
> 
> https://github.com/igvteam/igv/issues/283
> 
> https://github.com/igvteam/igv/pull/280
> 
> What you can do is to align the “tiling reads” to the contigs as bam file with `bwa mem` or `blasr`. And, you can use IGV to open the bam file, then you right-click the reads on the browser to get a menu. There is a menu item “ExtView” that can initiate a http request to the server (`main_loop.py`). You will have to configure IGV through `~/igv/prefs.properties` with a new option `EXTVIEW_URL=http://127.0.0.1:6502` so the IGV knows the server address and port to connect to.
> 
> There are also a couple of hard-coded parameters for adjusting the size of subgraphs. It might useful to visualize the graph in different scale sometimes.
> 
> I hope this help for now. I will try to work on polish the code a bit and write up the documentation.

**Comment by ghost** [comment]
Great, thanks a lot Jason! Sry for being so impatient, but that happens when you keep to produce such nice tools :-D
